src/main/java/org/apache/mesos/offer/constrain/RoleRule.java

```diff
@@ -0,0 +1,41 @@
+package org.apache.mesos.offer.constrain;
+
+import org.apache.mesos.Protos.Offer;
+import org.apache.mesos.Protos.Resource;
+
+/**
+ * Requires that the Offer's resources be assigned to a provided role.
+ */
+public class RoleRule implements PlacementRule {
```

Comment:
This RoleRule is a bit dangerous in my view.  I don't know of a use case for it.  Pre-reserved out of band resources are not currently supported by the rest of the Offer Evaluation stack.  If anyone actually tried to use this today, nothing would work.  I'm tempted to say we leave it out since it can't be used yet (or maybe ever) depending on whether we ever fully support static partitioning.